File: коды ВР8.cpp
#include <iostream>
#include <vector>
using namespace std;

int array_sum(vector<int> arr) {
    if (arr.empty()) {
        return 0;
    }
    int first = arr[0];
    arr.erase(arr.begin());
    return first + array_sum(arr);
}

// Пример использования
int main() {
    vector<int> nums = {1, 2, 3, 4, 5};
    cout << array_sum(nums) << endl;  // Вывод: 15
    return 0;
}

File: коды ВР8.java
import java.util.Arrays;

public class Main {
    public static int arraySum(int[] arr) {
        if (arr.length == 0) {
            return 0;
        }
        return arr[0] + arraySum(Arrays.copyOfRange(arr, 1, arr.length));
    }
    
    // Пример использования
    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 4, 5};
        System.out.println(arraySum(nums));  // Вывод: 15
    }
}
